diff --git a/resources/testharness.js b/resources/testharness.js
index f0c2463501..a42a47d293 100644
--- a/resources/testharness.js
+++ b/resources/testharness.js
@@ -2489,6 +2489,7 @@ policies and contribution forms [3].
         this.output_node = null;
         this.enabled = settings.output;
         this.phase = this.INITIAL;
+        this.tt_policy = null;
     }

     Output.prototype.INITIAL = 0;
@@ -2517,6 +2518,11 @@ policies and contribution forms [3].
             this.output_document = document;
         }
         this.phase = this.STARTED;
+        if (window.TrustedTypes) {
+            this.tt_policy = window.TrustedTypes.createPolicy('web-platform-tests-internal-unsafe', {
+                'createHTML': function(s) { return s; }
+            });
+        }
     };

     Output.prototype.resolve_log = function() {
@@ -2770,6 +2776,9 @@ policies and contribution forms [3].
         }
         html += "</tbody></table>";
         try {
+            if (this.tt_policy) {
+                html = this.tt_policy.createHTML(html);
+            }
             log.lastChild.innerHTML = html;
         } catch (e) {
             log.appendChild(document.createElementNS(xhtml_ns, "p"))
